### ISSUE TITLE
VIP: CREATE alternative to create_with_code_of

### ISSUE THREAD

**Issue by haydenadams** [ethereum/vyper#1362]
## General Idea
Right now `create_with_code_of(templateAddress)` creates a proxy delegatecall contract that uses the template as a library. Proxy contracts represent a tradeoff between storage at deployment and gas cost of calls. Right now there is no factory pattern available in Vyper for people who would prefer to reduce cost of calls. Having an alternate version that deploys a new contract with identical bytecode to the template would be useful. 

## Implementation
`create_with_code_of` is named confusingly, especially since it does not actually duplicate bytecode. I suggest two new functions:

`deploy_contract(template: address)` - duplicate template bytecode to new contract
`deploy_proxy_contract(template: address)`  - deploy new proxy contract using template as library

This is also a good time to figure out: https://github.com/ethereum/vyper/issues/773 and maybe keep https://github.com/ethereum/vyper/issues/649 in mind!

IMO it would be good to figure out the factory patterns better before the first release!

**Comment by jacqueswww** [comment]
@haydenadams We've already renamed it to `create_forwarder_to`. So we'd work of, of that as a name.

https://vyper.readthedocs.io/en/v0.1.0-beta.9/built-in-functions.html#create-forwarder-to

**Comment by haydenadams** [comment]
@jannikluhn 

> Just to eloborate the above copies the bytecode in it's entirety using CREATE (with full EXTCODE/EXTCODESIZE) usage

Yes, since removing the delegatecall will save gas for all transactions
